go/vt/tabletmanager/after_action.go

```diff
@@ -160,24 +161,30 @@ func (agent *ActionAgent) changeCallback(ctx context.Context, oldTablet, newTabl
 	var tabletControl *pbt.Shard_TabletControl
 	var blacklistedTables []string
 	var err error
+	var disallowQueryReason string
 	if allowQuery {
 		shardInfo, err = agent.TopoServer.GetShard(ctx, newTablet.Keyspace, newTablet.Shard)
 		if err != nil {
 			log.Errorf("Cannot read shard for this tablet %v, might have inaccurate SourceShards and TabletControls: %v", newTablet.Alias, err)
 		} else {
 			if newTablet.Type == pbt.TabletType_MASTER {
-				allowQuery = len(shardInfo.SourceShards) == 0
+				if allowQuery = len(shardInfo.SourceShards) == 0; !allowQuery {
+					disallowQueryReason = "old master is still in shard info"
+				}
```

Comment:
I would rewrite this as:
if len(shardInfo.SourceShards) > 0 {
  allowQuery = false
  disallowQueryReason = "old master is still in shard info"
}
I think it's more readable